nemo-physical/src/tabular/buffer/tuple_buffer.rs

```diff
@@ -477,18 +477,18 @@ impl TupleBuffer {
             return true;
         }
 
-        for pattern in &self.patterns {
+        'pattern: for pattern in &self.patterns {
             if !self.pattern_matches(pattern) {
                 continue;
             }
 
             let data_values = self.current_tuple_data_values.clone();
             for transformation in &pattern.transformations {
-                let value = transformation
-                    .program
-                    .evaluate_data(&data_values)
-                    .expect("should evaluate to a value");
-                self.current_tuple_data_values[transformation.position] = value;
+                if let Some(value) = transformation.program.evaluate_data(&data_values) {
+                    self.current_tuple_data_values[transformation.position] = value;
+                } else {
+                    continue 'pattern;
```

Comment:
I had to take a moment to think about why this would not evaluate to a value (when the data types do not match, e.g.) – a comment would be a good thing here